pkg/connector/client/confluence.go

```diff
@@ -341,3 +344,82 @@ func strToInt(s string) int {
 	}
 	return i
 }
+
+// WithConfluenceRatelimitData Per the docs: transient 5XX errors should be
+// treated as 429/too-many-requests if they have a retry header. 503 errors were
+// the only ones explicitly called out, but I guess it's possible for others too
+// https://developer.atlassian.com/cloud/confluence/rate-limiting/
+func WithConfluenceRatelimitData(resource *v2.RateLimitDescription) uhttp.DoOption {
+	return func(response *uhttp.WrapperResponse) error {
+		rateLimitData, err := helpers.ExtractRateLimitData(response.StatusCode, &response.Header)
+		if err != nil {
+			return err
+		}
+		resource = rateLimitData
+		return nil
+	}
+}
+
+func isRatelimited(
+	ratelimitStatus v2.RateLimitDescription_Status,
+	statusCode int,
+) bool {
+	return slices.Contains(
+		[]v2.RateLimitDescription_Status{
+			v2.RateLimitDescription_STATUS_OVERLIMIT,
+			v2.RateLimitDescription_STATUS_ERROR,
+		},
+		ratelimitStatus,
+	) || slices.Contains(
+		[]int{
+			http.StatusTooManyRequests,
+			http.StatusGatewayTimeout,
+			http.StatusServiceUnavailable,
+		},
+		statusCode,
+	)
+}
+
+// GetUsersFromSearch There are no official, documented ways to get lists of
+// users in Confluence. One way to get users is to issue a CQL search query with
+// no conditions. The documentation mentions that queries return "up to 10k"
+// users. So that may end up being a limitation of this approach.
+func (c *ConfluenceClient) GetUsersFromSearch(
+	ctx context.Context,
+	pageToken string,
+	pageSize int,
+) (
+	[]ConfluenceUser,
+	string,
+	*v2.RateLimitDescription,
+	error,
+) {
+	getUsersUrl, err := c.parse(
+		SearchUrlPath,
+		withLimitAndOffset(pageToken, pageSize),
```

Comment:
This works once you add `withQueryParameters(map[string]interface{}{"cql": "type=user"}),`